src/Components/Shelf/paginate.ts

```diff
@@ -0,0 +1,101 @@
+/**
+ * Sums all values, building from the previous
+ * @param xs list of values to compound
+ */
+export const compound = (xs: number[]): number[] => {
+  // @ts-expect-error  MIGRATE_STRICT_MODE
+  return xs.reduce((p, n) => p.concat((+p.slice(-1) || 0) + n), [])
+}
+
+/**
+ * Build an array of offsets
+ * @param viewport window width
+ * @param values array of cell widths
+ */
+export const chunk = ({
+  viewport,
+  values,
+}: {
+  viewport: number
+  values: number[]
+}) => {
+  return values.reduce(
+    (acc, value) => {
+      const prev = acc[acc.length - 1]
+      if (prev + value <= viewport) {
+        acc[acc.length - 1] += value
+        return acc
+      }
+      return [...acc, value]
+    },
+    [0]
+  )
+}
+
+/** Align last page to the left or right */
+export type CarouselPaginationAlignment = "left" | "right"
+
+export type CarouselPaginateBy = "page" | "cell"
+
+interface PaginateCarouselProps {
+  viewport: number
+  values: number[]
+  alignment?: CarouselPaginationAlignment
+  paginateBy?: CarouselPaginateBy
+}
+
+const paginateCarouselByCell = ({
+  values,
+  viewport,
+}: Pick<PaginateCarouselProps, "values" | "viewport">) => {
+  const sum = values.reduce((acc, curr) => acc + curr, 0)
+
+  // No need to paginate
+  if (sum <= viewport) {
+    return [0]
+  }
+
+  const offsets = compound(values)
+  const lastOffset = offsets[offsets.length - 1]
+  const offsetToLastCell = lastOffset - viewport + 100
```

Comment:
@dimatretyak I imported this file and added 100 to this. Only way it worked

Reply:
unfortunately, I don't think this is a working solution that will be applicable for all cases (not just search bar pills)

Reply:
if it's not difficult, can you please explain why exactly `100` is added here?

Reply:
Yes its not a solution. I am just showing you where I made a change to make it work. We need the right value for this.